openedx/core/djangoapps/edx_global_analytics/tests/test_utils.py

```diff
@@ -0,0 +1,157 @@
+"""
+Tests for edx_global_analytics application helper functions aka utils.
+"""
+
+import datetime
+
+from django.test import TestCase
+from django.utils import timezone
+
+from django.db.models import Q
+from django_countries.fields import Country
+
+from student.models import UserProfile
+from student.tests.factories import UserFactory
+
+from ..utils import fetch_instance_information, cache_instance_data
+
+
+class TestStudentsAmountPerParticularPeriod(TestCase):
+    """
+    Tests cover all methods, that have a deal with statistics calculation.
+    """
+    @staticmethod
+    def active_students_amount_default_data():
+        """
+        Default integration and particular data for active students amount functionality.
+        """
+        users_last_login = [
+            timezone.make_aware(datetime.datetime(2017, 5, 14, 23, 59, 59), timezone.get_default_timezone()),
+            timezone.make_aware(datetime.datetime(2017, 5, 15, 00, 00, 00), timezone.get_default_timezone()),
+            timezone.make_aware(datetime.datetime(2017, 5, 15, 23, 59, 59), timezone.get_default_timezone()),
+            timezone.make_aware(datetime.datetime(2017, 5, 16, 00, 00, 00), timezone.get_default_timezone()),
+            timezone.make_aware(datetime.datetime(2017, 5, 16, 00, 00, 01), timezone.get_default_timezone())
```

Comment:
Leading zeros don't give you any precision.
What's worse, leading zeros actually give you octal numbers. Not a disaster in this particular case, but certainly not what you intend.